backend/maint-scripts/update_scraper_version.py

```diff
@@ -174,24 +226,56 @@ def update_entries(
         else:
             offliner_definition = None
 
-        nb_schedules_modified = update_entries(
-            session,
-            model=Schedule,
-            offliner=offliner,
-            offliner_definition=offliner_definition,
-            image_tag=args.image_tag,
-            name_mappings=args.name_mappings,
-        )
-        logger.info(f"updated {nb_schedules_modified} schedule(s) ")
-
-        nb_requested_tasks_modified = update_entries(
-            session,
-            model=RequestedTask,
-            offliner=offliner,
-            offliner_definition=offliner_definition,
-            image_tag=args.image_tag,
-            name_mappings=args.name_mappings,
-        )
-        logger.info(f"updated {nb_requested_tasks_modified} requested tasks ")
+        if args.offliner_definition_version or args.image_tag or args.name_mappings:
+            nb_schedules_modified = update_entries(
+                session,
+                model=Schedule,
+                offliner=offliner,
+                offliner_definition=offliner_definition,
+                image_tag=args.image_tag,
+                name_mappings=args.name_mappings,
+            )
+            logger.info(f"updated {nb_schedules_modified} schedule(s) ")
+
+            nb_requested_tasks_modified = update_entries(
+                session,
+                model=RequestedTask,
+                offliner=offliner,
+                offliner_definition=offliner_definition,
+                image_tag=args.image_tag,
+                name_mappings=args.name_mappings,
+            )
+            logger.info(f"updated {nb_requested_tasks_modified} requested tasks ")
+
+        if args.similarity_data:
+            logger.info("Updating simlarity data")
+            similarity_data = [
+                SimilarityDataSchema.model_validate(data)
+                for data in json.loads(args.similarity_data.read())
+            ]
+            for offliner_definition in session.scalars(
+                select(OfflinerDefinition).where(
+                    OfflinerDefinition.offliner == args.offliner
+                )
+            ):
+                # update the spec for the offliner definition to use the new
+                # similarity data transformers list
+                offliner_spec = OfflinerSpecSchema.model_validate(
+                    offliner_definition.schema
+                )
+                offliner_spec.similarity_data = similarity_data
+                # update the db offliner definition and all of its associated recipes
+                offliner_definition.schema = offliner_spec.model_dump(mode="json")
+                nb_schedules_modified = update_entries(
```

Comment:
Looks like this is wrong, it is modifying the offliner definition used by schedules while we want to keep the ones currently used. 

We just want to update similarity data of every schedule (this is working well). And only once per schedule, not once per schedule and per offliner definition present in the DB.